### PR TITLE
Suggest an appropriate module name with the 'defprotocol' snippet

### PR DESCRIPTION
**Pull request by oo6**

### CODE REVIEW

File: apps/language_server/lib/language_server/providers/completion.ex [hunk 1]
```diff
@@ -587,6 +587,16 @@ defmodule ElixirLS.LanguageServer.Providers.Completion do
...
   end
 
+  defp snippet_for({"Kernel", "defprotocol"}, %{file_path: file_path})
+       when is_binary(file_path) do
+    "defprotocol #{suggest_module_name(file_path)}$1 do\n\t$0\nend"
+  end
+
+  defp snippet_for({"Kernel", "defimpl"}, %{file_path: file_path})
+       when is_binary(file_path) do
+    "defimpl #{suggest_module_name(file_path)}$1, for: $2 do\n\t$0\nend"
```
Comment (lukaszsamson) [inline apps/language_server/lib/language_server/providers/completion.ex:597]: This won’t likely be a good module name. The protocol is often from some other lib. How about suggesting ‘for’ instead?

Comment (oo6) [reply apps/language_server/lib/language_server/providers/completion.ex:597]: OK, I got it.

> How about suggesting ‘for’ instead?

If we use `defprotocol` and `defimpl` in one file like `Enumerable`, it doesn't make sense.
So, I delete the `defimpl` pattern match? Or do you have any suggestions?

**Comment by lukaszsamson** [comment]
@oo6 please look into failing tests

**Comment by oo6** [comment]
> @oo6 please look into failing tests

fixed

**Comment by lukaszsamson** [comment]
@oo6 Can you resolve merge conflicts?

**Comment by oo6** [comment]
> @oo6 Can you resolve merge conflicts?

Done.
By the way, the test failure doesn't seem to be related to my changes.

**Comment by lukaszsamson** [comment]
Thanks

**Pull request merged**